feat(temporary-shipping-address): add instruction on how to do it (#702)
front-commerce/developers.front-commerce.com: versioned_docs/version-2.x/advanced/checkout/temporary-shipping-address.mdx

---
sidebar_position: 2
title: Temporary shipping address
description:
  By default, Front-Commerce theme manages shipping addresses in the customer
  account address book. However, it is still possible to manage shipping
  addresses that will not be saved in the address book.
---

<p>{frontMatter.description}</p>

:::info

This feature is only available for Magento1 and Magento2 classic checkout.

Support for negociated quote will be available soon

:::

## Classic checkout

You can test these queries in the
[GraphQL Playground](/docs/2.x/magento2/add-new-attribute#discover-the-playground)

> This example set an address in customer's address book as the shipping address :

Reviewer: **nit:** no spaces before semicolumn

```suggestion
This example set an address in customer's address book as the shipping address:
```

```graphql
mutation {
  setCheckoutShippingInformation(
    cartId: "189"
    shippingMethod: { carrier_code: "flatrate", method_code: "flatrate" }
    shippingAddress: { id: 12 }
  ) {
    success
  }
}
```

> This example set a non saved address as the shipping address :

Reviewer: **nit:** I feel like the sentence turned out a bit weird

```suggestion
This example set an address that will not be saved as the shipping address:
```

```graphql
mutation {
  setCheckoutShippingInformation(
    cartId: "209"
    shippingMethod: { carrier_code: "flatrate", method_code: "flatrate" }
    shippingAddress: {
      address: {
        firstname: "Testing"
        lastname: "Front-Commerce"
        street: ["42 Street of test"]
        postcode: "31000"
        city: "Toulouse"
        country_id: "FR"
      }
    }
  ) {
    success
    errorMessage
  }
}
```

> To achieve this, you must override the
> `web/theme/pages/Checkout/stepsDefinition.js` file and remove the highlighted
> lines (this will disable address without ID protection for logged in customers)

Reviewer: **nit(non-blocking):** I think it could be clearer.
```suggestion
To achieve this, you will need to remove the check for addresses ids for logged in customers in the `web/theme/pages/Checkout/stepsDefinition.js`:
```
and combining this using a `diff` code block could help visualize this better as well.

I've no hard feelings about this change though.
